### DISCUSSION TITLE
ui.input -- can we change color for placeholder text, the text written and icons?

### DISCUSSION THREAD

**Discussion by Anindya088** [zauberzeug/nicegui#998] [Q&A] [answered]
### Question

Hi All,

I am trying to create a login card, with a background image which is quite dark. So I am trying to keep all white(ish) color for the texts and inputs. Ex: screenshot below:
![image](https://github.com/zauberzeug/nicegui/assets/65127974/b8f4ac72-89f9-4001-8d90-ac196b99e1d1)

The code I am using is this below:
```python
with ui.card().classes('bg-transparent'):
            ui.input("Username").props('label-color=cyan-8 clearable')
            ui.input('Password', password=True, password_toggle_button=True).props('label-color=cyan-8 clearable')
```

So I am using the label, together with the prop for label-color to change them.

So I need 3 helps basically:
1. Is there a way to change the colors for the placeholder text? Ex: if I use instead ```ui.input(placeholder="Username")```, can I format that color? 
2. How do I change the color of the actual text written? See the screenshot highlighted. 
3. How do I change the colors for the icons like the ```password_toggle_button```

My apologies for probably asking very basic stuffs, but I am still learning the basics, and thanks for all the help.

Regards,
Anindya

**Comment by falkoschindler** [comment, marked as answer]
Hi @Anindya088!

To answer questions 1 and 2: The placeholder text gets the same color like the written text, which can be adjusted with `.props('... input-class=text-red')`:
```py
i = ui.input('Password', password=True, password_toggle_button=True) \
    .props('label-color=cyan-8 clearable input-class=text-red')
```

To change the password toggle icon (question 3), you'd need to reach into the "append" slot which contains the icon:
```py
i.slots['append'].children[0].classes('text-blue-8')
```

Changing the color of the clear icon is a bit more tricky. But we can add a custom class to the input element and address it with custom CSS:
```py
ui.add_head_html('<style>.name-input button { color: red; }</style>')
ui.input('Name').props('clearable').classes('name-input')
```

Note that many Quasar elements have a prop "dark" to "Notify the component that the background is a dark color". This way you might not need to adjust the colors individually:
```py
with ui.card().classes('bg-gray-800'):
    ui.input('Password', placeholder='Start typing', password=True, password_toggle_button=True) \
        .props('label-color=cyan-8 clearable dark')
```

**Reply by rodja** [reply]
An alternative to defining a custom css would be to implement clearing yourself which gives you even more flexibility:

```py
with ui.input(value='custom clear button').classes('w-64') as i:
    ui.button(color='orange-8', on_click=lambda: i.set_value('')) \
        .props('icon=delete flat dense').bind_visibility_from(i, 'value')
```

**Reply by Anindya088** [reply]
Thanks so much, both. this works beautifully! I tried out the custom clearing button also and it's great too.
Must admit, there is so much to learn about the individual elements and how to control them :)

**Reply by rodja** [reply]
Year! Start nice and easy. Then go deeper and deeper. From our perspective, there is no end in sight 🤣
